### ISSUE TITLE
Please add `.IsNull` helper on handle structs to improve the usage syntax over `.Value == 0`

### ISSUE THREAD

**Issue by jnm2** [microsoft/CsWin32#206]
Instead of one of these:
```cs
if (PInvoke.SelectObject(bitmapDC, objectHandle).Value == 0)
```
```cs
if (PInvoke.SelectObject(bitmapDC, objectHandle).Equals(0))
```

I'd like to do something with the word 'null' in it since it's a handle:
```cs
if (PInvoke.SelectObject(bitmapDC, objectHandle).IsNull)
```

Alternatives that I would not prefer are ` == HGDIOBJ.Null` or `== default`.

**Comment by AArnott** [comment]
Similar to #122. Let's use the same pattern for both.

**Comment by AArnott** [comment]
#122 is similar, in that we want an `IsNull` property exposed there. But that's for a vtbl which clearly is null or non-null.

In this case, what you probably really want to know is whether the handle is *invalid*. Note some handles are invalid when set to `0`, while others are invalid at `-1`. Some are invalid with either value. So I think the new property should be called `IsInvalid`.

We're blocked on delivering this till [the metadata delivers a way](https://github.com/microsoft/win32metadata/issues/25#issuecomment-796621071) for us to know which values are in fact invalid.

**Comment by jnm2** [comment]
Actually, I'm specifically interested in `NULL` (0), not in handle validity, because this is how SelectObject signals an error.

https://docs.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-selectobject#return-value

> If an error occurs and the selected object is not a region, the return value is NULL.

**Comment by jnm2** [comment]
@AArnott Given my last comment, would you accept a PR for this?

**Comment by jnm2** [comment]
@AArnott I'm getting `isHandle == false` in DeclareTypeDefStruct for HGDIOBJ. Does that mean metadata is wrong? Should I make all typedef structs have IsNull or just handle structs?

**Comment by AArnott** [comment]
I would use option 4 from your list. Supporting down-level lang versions is a (low priority) goal and I want to avoid branching for different versions unless there's a really compelling reason, so going with the syntax that's supported by lower versions is preferred. That said, I can't see us supporting anything less than C# 7.1.

>  Does that mean metadata is wrong? 

No. That's probably just a side-effect of our non-support of `AlsoUsableForAttribute` (#199). If `HGDIOBJ` has that attribute, we might follow to the next struct and see if it has an `RIAAFree` attribute on it.
But then, some structs are handles that should never be freed, so I guess using `RIAAFree` to determine whether a typedef represents a handle isn't the greatest way.

> Should I make all typedef structs have IsNull or just handle structs?

Any `IntPtr` field-based typedef struct is probably a reasonable target, since as "pointers" `null` is a legit concept. If a typedef ever exists for what ought to be treated as `nint` we can deal with that later.

**Comment by jnm2** [comment]
Oh, hmm. I found this while searching for `== default` to make the change to `== IntPtr.Zero`:

https://github.com/microsoft/CsWin32/blob/e67e9c55652afa1139409790622c1176e730db02/src/Microsoft.Windows.CsWin32/Generator.cs#L1131

Should they stay consistent? Which way should they both be?

**Comment by AArnott** [comment]
`default` is fine.

**Comment by jnm2** [comment]
In that case, I'm already finished: #258
